### PR TITLE
[GE 1.12] Allow unknown fields

### PR DESCRIPTION
**Pull request by kdorosh**
related https://github.com/solo-io/gloo/issues/7293

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,8 +7,8 @@ replace (
 
 	// skv2 uses a newer version than the imported solo-kit version which causes issues. Replaces the version with the solo-kit version
 	github.com/pseudomuto/protoc-gen-doc => github.com/pseudomuto/protoc-gen-doc v1.0.0
-	k8s.io/apimachinery => k8s.io/apimachinery v0.20.9
-	k8s.io/client-go => k8s.io/client-go v0.20.9
+	k8s.io/apimachinery => k8s.io/apimachinery v0.21.4
```
Comment (nfuden) [inline go.mod:10]: Are there any breaking changes in the k8s upgrade?

Comment (kdorosh) [reply go.mod:10]: unfortunately there are no guarantees https://pkg.go.dev/k8s.io/apimachinery#readme-compatibility

however due to skv2 updates, there is no way to to get this to compile without adding a new skv2 branch just for this to track (seems like overkill given that no public apis were changed, no compilation errors, and this will only affect public consumers of this library)

i looked at the usages, we just use apimachinery to interact with the scheme and that code has not changed significantly.

tldr; i think the risk is near zero this breaks things, and if we did, it wouldn't affect our core product just people who update to a new version of solo-apis (this could always break things / be incompatible with their k8s version as we let it get stale)

further i think upgrading here is smart given we're already at 1.25 k8s upstream (k8s is much better about backwards compatability than forward for clients, of course)

**Review by nfuden — APPROVED** [review]
